### PR TITLE
Add support for mean sqrt pooling.

### PR DESCRIPTION
**Pull request by fkdosilovic**
#### Related PRs/Issues

Closes https://github.com/UKPLab/sentence-transformers/issues/2505

#### Description

- Add support for setting mean sqrt as pooling option
- Wrong pooling mode now throws a ValueError instead of invalidating an assert.
- Minor documentation improvements

### CODE REVIEW

File: sentence_transformers/models/Pooling.py [hunk 1]
```diff
@@ -22,6 +22,15 @@ class Pooling(nn.Module):
     :param pooling_mode_lasttoken: Perform last token pooling, see https://arxiv.org/abs/2202.08904 & https://arxiv.org/abs/2201.10005
     """
 
+    POOLING_MODES = (
+        "cls",
+        "lasttoken",
+        "max",
+        "mean",
+        "meansqrt",
```
Comment (tomaarsen) [inline sentence_transformers/models/Pooling.py:30]: I think we should use `mean_sqrt_len_tokens`. The inconsistency with `_` in `mean_sqrt_len_tokens` versus no `_` in `lasttoken` and `weightedmean` is pretty bothersome, so normally `meansqrt` would be a lot better, but we should probably stick with what we already had for backwards compatibility:

https://github.com/UKPLab/sentence-transformers/blob/e0d93bb377eca9e0c94da436f3f35c336e9c109d/sentence_transformers/models/Pooling.py#L95-L96

Comment (fkdosilovic) [reply sentence_transformers/models/Pooling.py]: Good catch. I missed this.

**Review by tomaarsen — APPROVED** [review]
Looks great!

**Comment by tomaarsen** [comment]
Thanks a bunch for the recent help with PRs & issues, I appreciate it!

**Comment by fkdosilovic** [comment]
> Thanks a bunch for the recent help with PRs & issues, I appreciate it!

Happy to help :)

**Pull request merged**